File: app/src/main/java/com/example/myapplication/ui/quiz/QuizFragment.java
package com.example.myapplication.ui.quiz;

import android.annotation.SuppressLint;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.example.myapplication.MainActivity;
import com.example.myapplication.R;
import com.example.myapplication.databinding.FragmentQuizBinding;

public class QuizFragment extends Fragment implements View.OnClickListener {

    private Button falseButton;
    private Button trueButton;
    private ImageButton nextButton;
    private ImageButton prevButton;
    private ImageView Image;
    private TextView questionTextView;
    private int correct = 0;
    // to keep current question track
    private int currentQuestionIndex = 0;

    private Question[] questionBank = new Question[] {
            // array of objects of class Question
            // providing questions from string
            // resource and the correct ans
            new Question(R.string.a, false),
            new Question(R.string.b, true),
            new Question(R.string.c, true),
            new Question(R.string.d, true),
            new Question(R.string.e, true),
            new Question(R.string.f, false),

    };

    private FragmentQuizBinding binding;
    @Override
    public View onCreateView(LayoutInflater inflater,ViewGroup container, Bundle args) {

        View view = inflater.inflate(R.layout.fragment_quiz, container, false);
        // setting up the buttons
        // associated with id
        falseButton = view.findViewById(R.id.false_button);
        trueButton = view.findViewById(R.id.true_button);
        nextButton = view.findViewById(R.id.next_button);
        prevButton = view.findViewById(R.id.prev_button);
        // register our buttons to listen to
        // click events
        questionTextView
                = view.findViewById(R.id.answer_text_view);
        Image = view.findViewById(R.id.myimage);
        falseButton.setOnClickListener(this);
        trueButton.setOnClickListener(this);
        nextButton.setOnClickListener(this);
        prevButton.setOnClickListener(this);


        return view;
    }
    @SuppressLint({"SetTextI18n", "NonConstantResourceId"})
    @Override
    public void onClick(View v)
    {
        // checking which button is
        // clicked by user
        // in this case user choose false
        switch (v.getId()) {
            case R.id.false_button:
                checkAnswer(false);
                break;

            case R.id.true_button:
                checkAnswer(true);
                break;

            case R.id.next_button:
                // go to next question
                // limiting question bank range
                if (currentQuestionIndex < 7) {
                    currentQuestionIndex
                            = currentQuestionIndex + 1;
                    // we are safe now!
                    // last question reached
                    // making buttons
                    // invisible
                    if (currentQuestionIndex == 6) {
                        questionTextView.setText(getString(
                                R.string.correct, correct));
                        nextButton.setVisibility(
                                View.INVISIBLE);
                        prevButton.setVisibility(
                                View.INVISIBLE);
                        trueButton.setVisibility(
                                View.INVISIBLE);
                        falseButton.setVisibility(
                                View.INVISIBLE);
                        if (correct > 3)

                            questionTextView.setText(
                                    "CORRECTNESS IS " + correct
                                            + " "
                                            + "OUT OF 6");
                            // showing correctness
                        //else
                            //Image.setImageResource(
                            //        R.drawable.resu);
                        // if correctness<3 showing sad emoji
                    }
                    else {
                        updateQuestion();
                    }
                }

                break;
            case R.id.prev_button:
                if (currentQuestionIndex > 0) {
                    currentQuestionIndex
                            = (currentQuestionIndex - 1)
                            % questionBank.length;
                    updateQuestion();
                }
        }
    }

    private void updateQuestion()
    {
        Log.d("Current",
                "onClick: " + currentQuestionIndex);

        questionTextView.setText(
                questionBank[currentQuestionIndex]
                        .getAnswerResId());
        // setting the textview with new question
        switch (currentQuestionIndex) {
            case 1:
                // setting up image for each
                // question
                //Image.setImageResource(R.drawable.f2);
                break;
            case 2:
                //Image.setImageResource(R.drawable.f3);
                break;
            case 3:
                //Image.setImageResource(R.drawable.f4);
                break;
            case 4:
                //Image.setImageResource(R.drawable.f5);
                break;
            case 5:
                //Image.setImageResource(R.drawable.f6);
                break;
            case 6:
                //Image.setImageResource(R.drawable.f7);
                break;
            case 7:
                //Image.setImageResource(R.drawable.f1);
                break;
        }
    }
    private void checkAnswer(boolean userChooseCorrect)
    {
        boolean answerIsTrue
                = questionBank[currentQuestionIndex]
                .isAnswerTrue();
        // getting correct ans of current question
        int toastMessageId;
        // if ans matches with the
        // button clicked

        if (userChooseCorrect == answerIsTrue) {
            toastMessageId = R.string.correct_answer;
            correct++;
        }
        else {
            // showing toast
            // message correct
            toastMessageId = R.string.wrong_answer;
        }

        Toast.makeText(getActivity(), toastMessageId,
                        Toast.LENGTH_SHORT)
                .show();
    }
}

File: app/src/main/java/com/example/myapplication/ui/quiz/QuizFragment.kt
package com.example.myapplication.ui.quiz

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.fragment.app.Fragment
import com.example.myapplication.R
import com.example.myapplication.databinding.FragmentQuizBinding

class QuizFragment : Fragment(), View.OnClickListener {
    private var falseButton: Button? = null
    private var trueButton: Button? = null
    private var nextButton: ImageButton? = null
    private var prevButton: ImageButton? = null
    private var Image: ImageView? = null
    private var questionTextView: TextView? = null
    private var correct = 0

    // to keep current question track
    private var currentQuestionIndex = 0
    private val questionBank = arrayOf( // array of objects of class Question
        // providing questions from string
        // resource and the correct ans
        Question(R.string.a, false),
        Question(R.string.b, true),
        Question(R.string.c, true),
        Question(R.string.d, true),
        Question(R.string.e, true),
        Question(R.string.f, false)
    )
    private val binding: FragmentQuizBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        args: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_quiz, container, false)
        // setting up the buttons
        // associated with id
        falseButton = view.findViewById(R.id.false_button)
        trueButton = view.findViewById(R.id.true_button)
        nextButton = view.findViewById(R.id.next_button)
        prevButton = view.findViewById(R.id.prev_button)
        // register our buttons to listen to
        // click events
        questionTextView = view.findViewById(R.id.answer_text_view)
        Image = view.findViewById(R.id.myimage)
        falseButton?.setOnClickListener(this)
        trueButton?.setOnClickListener(this)
        nextButton?.setOnClickListener(this)
        prevButton?.setOnClickListener(this)
        return view
    }

    @SuppressLint("SetTextI18n", "NonConstantResourceId")
    override fun onClick(v: View) {
        // checking which button is
        // clicked by user
        // in this case user choose false
        when (v.id) {
            R.id.false_button -> checkAnswer(false)
            R.id.true_button -> checkAnswer(true)
            R.id.next_button ->                 // go to next question
                // limiting question bank range
                if (currentQuestionIndex < 7) {
                    currentQuestionIndex = currentQuestionIndex + 1
                    // we are safe now!
                    // last question reached
                    // making buttons
                    // invisible
                    if (currentQuestionIndex == 6) {
                        questionTextView!!.text = getString(
                            R.string.correct, correct
                        )
                        nextButton!!.visibility = View.INVISIBLE
                        prevButton!!.visibility = View.INVISIBLE
                        trueButton!!.visibility = View.INVISIBLE
                        falseButton!!.visibility = View.INVISIBLE
                        if (correct > 3) questionTextView!!.text = ("CORRECTNESS IS " + correct
                                + " "
                                + "OUT OF 6")
                        // showing correctness
                        //else
                        //Image.setImageResource(
                        //        R.drawable.resu);
                        // if correctness<3 showing sad emoji
                    } else {
                        updateQuestion()
                    }
                }
            R.id.prev_button -> if (currentQuestionIndex > 0) {
                currentQuestionIndex = ((currentQuestionIndex - 1)
                        % questionBank.size)
                updateQuestion()
            }
        }
    }

    private fun updateQuestion() {
        Log.d(
            "Current",
            "onClick: $currentQuestionIndex"
        )
        questionTextView?.setText(
            questionBank[currentQuestionIndex]
                .answerResId
        )
        when (currentQuestionIndex) {
            1 -> {}
            2 -> {}
            3 -> {}
            4 -> {}
            5 -> {}
            6 -> {}
            7 -> {}
        }
    }

    private fun checkAnswer(userChooseCorrect: Boolean) {
        val answerIsTrue = questionBank[currentQuestionIndex]
            .isAnswerTrue
        // getting correct ans of current question
        val toastMessageId: Int
        // if ans matches with the
        // button clicked
        if (userChooseCorrect == answerIsTrue) {
            toastMessageId = R.string.correct_answer
            correct++
        } else {
            // showing toast
            // message correct
            toastMessageId = R.string.wrong_answer
        }
        Toast.makeText(
            activity, toastMessageId,
            Toast.LENGTH_SHORT
        )
            .show()
    }
}